packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -127,13 +141,21 @@ impl Abigen {
                            let provider = self.wallet.get_provider()?;
                            wallet.set_provider(provider.clone());
 
-                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet })
+                           Ok(Self { contract_id: self.contract_id.clone(), wallet: wallet, logs_lookup: self.logs_lookup.clone() })
                         }
 
-                        pub fn methods(&self) -> #methods_name {
-                            #methods_name { contract_id: self.contract_id.clone(), wallet: self.wallet.clone() }
+                        pub fn logs_with_type<D: Tokenizable + Parameterize>(&self, receipts: &[Receipt]) -> Result<Vec<D>, SDKError> {
+                            extract_and_parse_logs(&self.logs_lookup, receipts)
                         }
 
+                        #print_logs
+
+                        pub fn methods(&self) -> #methods_name {
+                            #methods_name {
+                                contract_id: self.contract_id.clone(),
+                                wallet: self.wallet.clone(),
+                            }
+                        }
```

Comment:
cosmetics :smile: I would put `#print_logs` at the end.
```suggestion
                        pub fn methods(&self) -> #methods_name {
                            #methods_name {
                                contract_id: self.contract_id.clone(),
                                wallet: self.wallet.clone(),
                            }
                        }

                        #print_logs                        
```